### PR TITLE
Add ConfirmModal component

### PR DESCRIPTION
**Pull request by ShrawanBK**

### CODE REVIEW

File: src/components/ConfirmModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import { IoMdClose } from 'react-icons/io';
+
+import Button from '../Button';
+import BodyBackdrop from '../BodyBackdrop';
+import { UiMode } from '../ThemeContext';
+import { useThemeClassName } from '../../hooks';
+
+import styles from './styles.css';
+
+export interface ConfirmModalProps {
```
Comment (AdityaKhatri) [inline src/components/ConfirmModal/index.tsx]: If this overlaps with props of Modal component, you need to create the props by extending from props of Modal component.
See Accordion's prop definition.

Comment (ShrawanBK) [reply src/components/ConfirmModal/index.tsx]: okay !!

**Review by AdityaKhatri — APPROVED** [review]

File: src/components/Button/index.tsx [hunk 2]
```diff
@@ -141,17 +141,17 @@ function Button<N extends number | string | undefined>(props: ButtonProps<N>) {
...
             {children && (
                 <div className={_cs(childrenClassName, styles.children)}>
-                    { children }
+                    { children}
                 </div>
             )}
             {actions && (
                 <div className={_cs(actionsClassName, styles.actions)}>
-                    { actions }
+                    { actions}
```
Comment (tnagorra) [inline src/components/Button/index.tsx]: Why change the spacings?

File: src/components/ConfirmButton/index.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+import Button, { ButtonProps } from '../Button';
+
+import Modal from '../Modal';
+
+import styles from './styles.css';
+
+export interface ConfirmButtonProps<N extends number | string | undefined> extends ButtonProps<N> {
+    confirmLabel?: string;
+    cancelLabel?: string;
```
Comment (tnagorra) [inline src/components/ConfirmButton/index.tsx]: These should also be `React.ReactNode`

File: src/components/ConfirmButton/index.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    onConfirm: () => void,
+    confirmButtonClassName?: string,
+    cancelButtonClassName?: string,
+    confirmationHeader?: ReactNode,
+    confirmationMessage?: string,
```
Comment (tnagorra) [inline src/components/ConfirmButton/index.tsx]: These should also be `React.ReactNode`

File: src/components/ConfirmButton/index.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    actionButtonsClassName?: string,
+    children?: ReactNode;
+}
+
+function ConfirmButton<N extends number | string | undefined>(props: ConfirmButtonProps<N>) {
+    const {
+        confirmationHeader = 'Confirm Now !',
```
Comment (tnagorra) [inline src/components/ConfirmButton/index.tsx]:
```suggestion
        confirmationHeader = 'Confirmation',
```

File: src/components/ConfirmButton/index.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        confirmationMessage = 'Are you sure?',
+        onConfirm,
+        confirmButtonClassName,
+        cancelButtonClassName,
+        actionButtonsClassName,
+        confirmLabel = 'Confirm',
+        cancelLabel = 'Cancel',
+        children,
```
Comment (tnagorra) [inline src/components/ConfirmButton/index.tsx]: No need to capture `children` and pass it explicitly

File: src/components/ConfirmButton/index.tsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+                                name="cancel-button"
+                                onClick={handleConfirmModalClose}
+                            >
+                                {cancelLabel}
+                            </Button>
+                            <Button
+                                className={_cs(styles.actionButton, confirmButtonClassName)}
+                                name="confirm-button"
+                                onClick={handleConfirmModalConfirm}
+                                variant="primary"
```
Comment (tnagorra) [inline src/components/ConfirmButton/index.tsx:75]: Add `autoFocus` here

File: src/components/ConfirmButton/index.tsx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+import Button, { ButtonProps } from '../Button';
+
+import Modal from '../Modal';
+
+import styles from './styles.css';
+
+export interface ConfirmButtonProps<N extends number | string | undefined> extends ButtonProps<N> {
+    confirmLabel?: string;
+    cancelLabel?: string;
+    onConfirm: () => void,
```
Comment (tnagorra) [inline src/components/ConfirmButton/index.tsx:13]: We may need a onCancel prop as well. (that can be optional)

File: src/components/Button/index.tsx (hunk 2 above)
Comment (tnagorra) [reply src/components/Button/index.tsx]: Either use `{icons}` or `{ icons }`, `{ icons}` is not consistent.

File: src/stories/ConfirmButton.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    <h2>Confirm Action?</h2>
+);
+
+const Template = (args: ConfirmButtonProps) => (
+    <ConfirmButton
+        {...args}
+        name="confirm-button"
+        confirmationHeader={ConfirmationHeader}
+        confirmationMessage="Confirmation message"
+        onConfirm={() => console.log('confirmed')}
```
Comment (tnagorra) [inline src/stories/ConfirmButton.stories.tsx]: You don't need to add handler on stories. Storybook automatically injects it for you!

**Review by tnagorra — APPROVED** [review]

File: src/components/ConfirmButton/index.tsx [hunk 10]
```diff
@@ -0,0 +1,91 @@
+import React, { ReactNode, useState, useCallback, Children } from 'react';
```
Comment (frozenhelium) [inline src/components/ConfirmButton/index.tsx]: Children defined but never used

**Review by frozenhelium — APPROVED** [review]

**Pull request merged**